### ISSUE TITLE
write_gml should raise warning or errors when booleans / invalid characters are written

### ISSUE THREAD

**Issue by hlin117** [networkx/networkx#1985]
The following lines of code don't work, due to GML specifications [here](http://www.fim.uni-passau.de/index.php?id=17297&L=1). 

```
import networkx as nx
G = nx.DiGraph()
G.add_node(1, attr_dict={"small": True})
nx.write_gml(G, "small.gml")
nx.read_gml("small.gml")
```

`read_gml` currently cannot parse the output file, because the output does not treat True (or any boolean values) as an ascii string:

In `small.gml`:

```
graph [
  directed 1
  node [
    id 0
    label 1
    small True
  ]
]
```

I think the `nx.write_gml(G, "small.gml")` should raise a warning if boolean values are written to an output file. In any case, some notification to a user that boolean values will not be able to be written correctly is preferable.

Related: #1386. This mentions that you cannot write underscores in GML.

**Comment by yash14123** [comment]
I am on it. :)

**Comment by theaverageguy** [comment]
Please check the PR.

**Comment by hlin117** [comment]
Please state the PR number here, @theaverageguy. Thanks!

**Comment by hlin117** [comment]
@theaverageguy: I still don't see your PR. Can you please reference your PR number to this issue?

**Comment by Michael-E-Rose** [comment]
According to @theaverageguy's [fork](https://github.com/theaverageguy/networkx/commits/master), there was no activity since Feb 28

**Comment by hagberg** [comment]
I suppose the right thing to do is raise an error when attempting to write data that isn't GML spec.

**Comment by hlin117** [comment]
Related: Trying to write underscores - or other not-allowed ascii characters into GML. (Issue #1386) I've updated the issue to reflect this.

**Comment by hlin117** [comment]
@hagberg I actually think that it's a bit radical to not allow booleans in the output file. Booleans are, after all, one of the primary data primitives. Maybe instead we should allow an option to substitute booleans with `1` or `0`. As for invalid ascii characters - underscores included - we might want to raise a warning, and suggest to the user to rename these variables.

Or even hold a dictionary of key remappings?

**Comment by hagberg** [comment]
It's fine with me to allow some options.  But the downside is that the function may get complicated to maintain and use (see some of the other output format code for examples of complexity).  If we check all of the data for validity the performance might be impacted too.

**Comment by dschult** [comment]
This might be related to #2118 . That asks for read_gml to handle bools

**Comment by dschult** [comment]
Fixed by #2513
